src/main/java/janet/TaskList.java

```diff
@@ -0,0 +1,100 @@
+package janet;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+/**
+ * Represents the list of tasks tracked by Janet.
+ */
+public class TaskList implements Serializable {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Class constructor which initialises a blank TaskList.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task to be added
+     */
+    public void add(Task task) {
+        assert(task != null);
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes the task at the specified index from the list, and returns the deleted task.
+     *
+     * @param index Index of the task in the list of tasks
+     * @return The task that has been deleted
+     */
+    public Task delete(int index) {
+        assert(index <= tasks.size());
+        return tasks.remove(index);
+    }
+
+    /**
+     * Returns the number of tasks in the task list.
+     *
+     * @return Number of tasks in list
+     */
+    public int size() {
```

Comment:
Would it be better to name this method getSize so that the function of the method is better understood?